packages/legacy/src/queries/secrets.ts

```diff
@@ -13,11 +15,69 @@ export const useSecretQuery = (
   const secretResource = createSecretResource(namespace);
   return useMockableQuery<ISecret>(
     {
-      queryKey: ['secrets', secretName],
+      queryKey: ['secret', secretName],
       queryFn: async () => await consoleFetchJSON(secretResource.namedPath(secretName)),
       refetchInterval: usePollingContext().refetchInterval,
       enabled: !!secretName,
     },
-    secretName ==='secure' ? MOCK_SECRET_SECURE : MOCK_SECRET_INSECURE
+    secretName === 'secure' ? MOCK_SECRET_SECURE : MOCK_SECRET_INSECURE
   );
 };
+
+// function decodeSecretData (secret: ISecret) {
+//   if (secret.data) {
+//     for (const [key, value] of Object.entries(secret.data)) {
+//       secret.data[key] = atob(value);
+//     }
+//   }
+//   return secret;
+// }
+
+export const useSecretsQuery = (
+  secretNames: string[] | null,
+  namespace: string,
+  // decodeData: boolean = false,
+): UseQueryResult<IKubeList<ISecret>> => {
+  const filterList = useCallback(
+    (data: IKubeList<ISecret>): IKubeList<ISecret> => ({
+      ...data,
+      items: data.items.filter((d) => secretNames.includes((d.metadata as IMetaObjectMeta)?.name)),
+    }),
+    [secretNames]
+  );
+
+  // const filterAndDecode = useCallback(
+  //   (data: IKubeList<ISecret>): IKubeList<ISecret> => {
+  //     const filtered = filterList(data);
+  //     filtered.items = filtered.items.map(decodeSecretData)
+  //     return filtered;
+  //   },
+  //   [filterList]
+  // );
+
+  const secretResource = createSecretResource(namespace);
```

Comment:
it's up to you but k8s API (see `packages/forklift-console-plugin/src/utils/fetch.ts`)  seems more likely to be re-used in the future.

Reply:
But in the legacy package?  I was just sticking to the original query's style.

Reply:
right.  `legacy` should reference methods from `forklift-console-plugin`.